src/Face.cuh

```diff
@@ -19,29 +19,61 @@ class MapNode;
 
 class Polyhedron;
 
-/// Object describing a polyhedron face
+/**
+ * Object describing a polyhedron face
+ *
+ * @warning <b>Before using the class, please, carefully read documentation for both copy and move constructors and
+ *      assignment operators!</b>
+ */
 class Face
 {
 public:
     /**
      * Creates a `Face` object
      *
-     * @param id Identifier of the polyhedron face
      * @param vertices Array of polyhedron vertices that belong to the face.
      *      Must be ordered in a special way (see note below)
      * @param n_of_vertices Number of vertices on the face
-     * @param node Some node laying on the face
      *
      * @note The vertices order. Looking on a face <b>from outside</b> the polyhedron, some vertex (let's call it A)
      * must be saved to vertices[0]. It's neighbour clockwise - vertex B (B is next to A clockwise) must be saved to
      * vertices[1] and so on. Assuming there are N vertices in total, A's neighbors are B clockwise and
      * X counterclockwise, X must be saved to vertices[N - 2], and A must be saved <b>again</b> to vertices[N - 1]
      */
-    __device__ Face(int id, const SpacePoint *vertices, int n_of_vertices);
+    __device__ Face(const SpacePoint *vertices, int n_of_vertices);
+
+    /**
+     * `Face` object copy assignment operator
+     *
+     * @warning When copying `Face` object, its `node` field is <b>not</b> copied, but is set to `nullptr`, which means
+     *      the new `Face` won't have a node attached. Please, be careful
+     */
+    __device__ Face &operator=(const Face &other);
```

Comment:
Codacy found an issue: ['Face::operator=' should return 'Face &'.](https://app.codacy.com/gh/physarumAdv/minds_crawl/pullRequest?prid=5984297)

Reply:
Well, to my mind, it does return what it should. Looks like something is wrong, probably because we are trying to analyze CUDA code as it is C++. I will contact Codacy Support and ask about this